components/cam/src/physics/cam/shoc.F90

```diff
@@ -2864,9 +2868,17 @@ subroutine adv_sgs_tke(nlev, shcol, dtime, shoc_mix, wthv_sec, &
 
   real(rtype) :: Ck, Cs, Ce, Ce1, Ce2, Cee
 
+#ifdef SCREAM_CONFIG_IS_CMAKE
+  if (use_cxx) then
+     call adv_sgs_tke_f(nlev, shcol, dtime, shoc_mix, wthv_sec, &
+          sterm_zt, tk, tke, a_diss)
+     return
+  endif
+#endif
+
   Cs=0.15_rtype
   Ck=0.1_rtype
-  Ce=Ck**3/Cs**4
```

Comment:
You should use `cube` and `pow<4>` for this, not `bfb_pow`. There's an important difference in the numerics of `pow` vs the others.

Reply:
Btw, I don't know if anybody's impl'ed `pow<int>(.)` yet, but it's come up before and should be impl'ed. Can default to standard `pow` but then specialize for 2 (`x*x`), 3 (`x*x*x`), 4 (`pow<2>(pow<2>(x))`), with BFB wrappers for F90.

Reply:
We did an integer power in Hommexx, as part of our bfb utilities. At some point, it should be ported to Ekat.

Reply:
Should we stick these into Ekat before this PR goes in? Or should we locally define macros for now? Or...?

Reply:
I have been working on this PR and `pow` functions are the only one which are causing NBFB differences on GPU (as expected). Everything is working fine on the CPU side.
I am okay with waiting if that will be a cleaner solution and I think it will be. Other option is to use non-pow functions/Macros for now and switch to Ekat provided functions later.